hal/src/argon/platform_ncp_argon.cpp

```diff
@@ -75,16 +75,16 @@ __attribute__((optimize("O0"))) int platform_ncp_update_module(const hal_module_
     SPARK_ASSERT(ncpClient);
     CHECK(ncpClient->on());
     // we pass only the actual binary after the module info and up to the suffix
-    const uint8_t* start = (const uint8_t*)module->info;
+    const uint8_t* start = (const uint8_t*)&module->info;
```

Comment:
This file is missing changes to `OtaUpdateSourceStream` and other changes. `platform_ncp_update_module` and `platform_ncp_fetch_module_info` should look almost identical to `develop` with exception of version caching.

Reply:
Added in changes as per develop. Had to add `system_cache.cpp` and `system_cache.h` files as well to serve the new changes.